pkg/distributor/distributor.go

```diff
@@ -732,6 +734,56 @@ func (d *Distributor) prePushForwardingMiddleware(next push.Func) push.Func {
 	}
 }
 
+// instanceLimitsMiddleware checks for instance limits and rejects request if this instance cannot process it at the moment.
+func (d *Distributor) instanceLimitsMiddleware(next push.Func) push.Func {
+	return func(ctx context.Context, req *mimirpb.WriteRequest, callerCleanup func()) (*mimirpb.WriteResponse, error) {
+		// We will report *this* request in the error too.
```

Comment:
I know that this is just copied, but I don't understand what this comment is telling me :) which error?

Reply:
It's actually talking about error that THIS request can trigger. (How do I know? I wrote that comment originally.) But we don't actually report number of inflight requests anymore, so this comment seems bit obsolete.